### DISCUSSION TITLE
Create user account for every visitor without registeration and update this anonymous user when they actually signup

### DISCUSSION THREAD

**Discussion by rhazegh** [nextauthjs/next-auth#1507] [Ideas]
I would like to remove the friction of creating an account before the user can use a service. The way I imagine this would work is this:

1. User visits a route on the service and chooses to start trying out the service
2. At this point, a unique cookie is created on the browser and also a user account is created in the back-end that is associated with this cookie (this user will have limited access to the service).
3. Every time the user comes back, they will be identified by this cookie. 
4. If at some point they decide they want to register for an account, they provide their email address and the account that is already created in the back-end for them is updated with this email address.
5. From this point on, the user can login with their email address.

#### - If not, do you think it is feasible to add this as a new feature?

**Comment by luisrudge** [comment]
@rhazegh I need something like this as well. Did you figure this out?

**Comment by rhazegh** [comment]
@luisrudge I ended up implementing it from scratch and used https://www.npmjs.com/package/next-iron-session  for session management.
Unfortunately it's a lot of work but I had to customize it to the specific requirements of the project. Best of luck to you.

**Comment by luisrudge** [comment]
@rhazegh thanks!
